src/Adapter/Shop/MaintenanceConfiguration.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/*
+ * 2007-2017 PrestaShop
+ * 
+ * NOTICE OF LICENSE
+ * 
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ * 
+ * DISCLAIMER
+ * 
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ * 
+ *  @author PrestaShop SA <contact@prestashop.com>
+ *  @copyright  2007-2017 PrestaShop SA
+ *  @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *  International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Shop;
```

Comment:
Does this file should not be placed in Core instead of Adapter?

Reply:
We can debate about this, but currently I followed what has been done on the other PRs.
All classes implementing `DataConfigurationInterface` can be found in the Adapter folder, and I followed that 'rule'.

Reply:
I can reply to that question too: I have some ideas about how to "improve" the way we setup all configuration pages (almost 80% of back office) *but* I need to experiment a little bit more before put the "final" implementations into Core namespace. For me, everything from Adapter is not part of our public API and PrestaShop developers should'nt use them. Maybe should we formalize some experiments with `@internal` flag like in Symfony project, I don't know...